docs/src/test/java/sample/javadsl/RunnableGraphStreamWrapperActor.java

```diff
@@ -0,0 +1,194 @@
+
+        Graph<UniformFanOutShape<EventMessage, EventMessage>, NotUsed> eventTypePartition = Partition.create(3, (Function<EventMessage, Object>) tuple -> {
+            switch (tuple.message) {
+                case "EVENT_TYPE_1":
+                    return 1;
+                case "EVENT_TYPE_2":
+                    return 2;
+                default:
+                    return 0;
+            }
+        });
+
+        // Build the Partition Graph
+        Graph<ClosedShape, Consumer.Control> completionStageGraph = GraphDSL.create(sourceStream, (builder, sourceShape) -> {
+
+            UniformFanOutShape<EventMessage, EventMessage> eventTypeFanOut = builder.add(eventTypePartition);
+            Outlet<EventMessage> unknownEventOutlet = eventTypeFanOut.out(0);
+            Outlet<EventMessage> firstPartitionOutlet = eventTypeFanOut.out(1);
+            Outlet<EventMessage> secondPartitionOutlet = eventTypeFanOut.out(2);
+
+            builder.from(sourceShape)
+                    .toFanOut(eventTypeFanOut)
+                        .from(unknownEventOutlet)
+                            .to(builder.add(commitOnlySink))
+                        .from(firstPartitionOutlet)
+                            .via(builder.add(firstPartitionFlow))
+                            .via(builder.add(produceMessageFlow))
+                            .to(builder.add((producerCommitSink)))
+                        .from(secondPartitionOutlet)
+                            .via(builder.add(secondPartitionFlow))
+                            .to(builder.add(commitOnlySink));
+
+            return ClosedShape.getInstance();
+        });
+
+        //#errorHandlingClosedRunnableGraph
+        // Self kill this actor so that BackoffSupervisor starts this actor(pipeline/stream) again!
+        RunnableGraph.fromGraph(completionStageGraph)
+                .run(materializer)
+                .isShutdown()
+                .whenComplete((done, throwable) -> {
+                    getSelf().tell(PoisonPill.getInstance(), getSelf());
+                });
+        //#errorHandlingClosedRunnableGraph
+
+    }
+
+    private EventMessage processMessage(EventMessage tuple2) {
+        // main business logic to process message
+        return tuple2;
+    }
+
+    private ProducerMessage.Message<String, String, ConsumerMessage.Committable> produceCommittableEvent(EventMessage eventTuple, String topic) {
+        return new ProducerMessage.Message<>(new ProducerRecord<String, String>(topic, "Produce New Message"), eventTuple.committableMessage.committableOffset());
+    }
+
+    /**
+     * Custom decider to decide the Supervision strategy.
+     */
+    private Function<Throwable, Supervision.Directive> stopOnSpecificExceptionDecider(ActorRef actorRef) {
+        return exception -> {
+
+            if (exception instanceof IllegalStateException) { // Any specific exception
+                return Supervision.stop(); // BackoffSupervisor will start again!
```

Comment:
perhaps clarify with a comment that this will not stop the actor, but it will stop the stream, and `whenComplete` will stop the actor